3DAmsterdam/Assets/Amsterdam3D/Scripts/OBJLoader/ObjStringLoader.cs

```diff
@@ -30,64 +37,113 @@ private void Start()
 		}
+
+		/// <summary>
+		/// Start the parsing of the OBJ, fetching the obj and mtl strings from Javascript
+		/// </summary>
 		public void LoadOBJFromJavascript()
 		{
 			StartCoroutine(ParseOBJFromString(JavascriptMethodCaller.FetchOBJDataAsString(), JavascriptMethodCaller.FetchMTLDataAsString()));
 		}
+
+		/// <summary>
+		/// Start the parsing of OBJ and MTL strings
+		/// </summary>
+		/// <param name="objText">The OBJ string data</param>
+		/// <param name="mtlText">The MTL string data</param>
+		/// <returns></returns>
 		private IEnumerator ParseOBJFromString(string objText, string mtlText = "")
 		{
-			//Display loading message covering entire screen
-			loadingObjScreen.ShowMessage("Loading " + objModelName + "...");
-			yield return new WaitForEndOfFrame();
-			yield return new WaitForSeconds(0.1f);
+			//Create a new gameobject that parses OBJ lines one by one
+			var newOBJLoader = new GameObject().AddComponent<ObjLoad>();
+			float remainingLinesToParse;
+			float totalLines;
+			float percentage;
 
-			var newOBJ = new GameObject().AddComponent<ObjLoad>();
+			//Parse the mtl file, filling our material library
 			if (mtlText != "")
-				newOBJ.SetMaterialData(mtlText);
+			{
+				newOBJLoader.SetMaterialData(ref mtlText);
+				remainingLinesToParse = newOBJLoader.ParseNextMtlLines(1);
+				totalLines = remainingLinesToParse;
+
+				loadingObjScreen.ShowMessage("Materialen worden geladen...");
+				while (remainingLinesToParse > 0)
+				{
+					remainingLinesToParse = newOBJLoader.ParseNextMtlLines(maxLinesPerFrame);
+					percentage = 1.0f - (remainingLinesToParse / totalLines);
+					loadingObjScreen.ProgressBar.Percentage(percentage/100.0f); //Show first percent
+					yield return null;
+				}
+			}
 
-			newOBJ.SetGeometryData(objText);
-			newOBJ.Build(defaultLoadedObjectsMaterial);
+			//Parse the obj line by line
+			newOBJLoader.SetGeometryData(ref objText);
+			loadingObjScreen.ShowMessage("Objecten worden geladen...");
+			remainingLinesToParse = newOBJLoader.ParseNextObjLines(1);
+			totalLines = remainingLinesToParse;
+			while (remainingLinesToParse > 0)
+			{
+				remainingLinesToParse = newOBJLoader.ParseNextObjLines(maxLinesPerFrame);
+				percentage = 1.0f - (remainingLinesToParse / totalLines);
+				loadingObjScreen.ProgressBar.SetMessage(Mathf.Round(percentage * 100.0f) + "%");
+				loadingObjScreen.ProgressBar.Percentage(percentage);
+				yield return null;
+			}
+
+			newOBJLoader.Build(defaultLoadedObjectsMaterial);
 			
 			//Make interactable
-			newOBJ.transform.Rotate(0, 90, 0);
-			newOBJ.transform.localScale = new Vector3(1.0f, 1.0f, -1.0f);
-;			newOBJ.name = objModelName;
-			newOBJ.gameObject.AddComponent<Draggable>();
-			newOBJ.gameObject.AddComponent<MeshCollider>().sharedMesh = newOBJ.GetComponent<MeshFilter>().sharedMesh;
+			newOBJLoader.transform.Rotate(0, 90, 0);
+			newOBJLoader.transform.localScale = new Vector3(1.0f, 1.0f, -1.0f);
```

Comment:
Waarom local scale? Ik zie eigenlijk geen situatie waar de lossy scale niet gewoon 1:1 moet zijn met de obj file. (Tenzij we nog een scale optie toevoegen aan obj files op dezelfde manier als we een basisvorm scalen).

Reply:
Daar spiegel in nu de geimporteerde OBJ mee ( die kwamen gespiegeld in beeld).
Niet heel fraai inderdaad  :]  Zal kijken of ik dat tijdens het parsen van de verts kan doen!